meeko/flexibility.py

```diff
@@ -5,275 +5,269 @@
-                self.set_graph_root(model, root_atom_index) # finds root if root_atom_index==None
-                self.add_flex_model(model, score=True, initial_score=bond_break_score)
-
-        self.select_best_model()
-
-        # clean up
-        del self._frozen_bonds
-        del self.flexibility_models
-        return self.setup
-
-
-    def select_best_model(self):
-        """
-        select flexibility model with best complexity score
-        """
-        if len(self.flexibility_models) == 1: # no macrocyle open rings
-            best_model = list(self.flexibility_models.values())[0]
+def _calc_max_weighted_depth(model, seed_node, bonds_to_break, visited=None, depth=0):
+    glue_atoms = []
+    for (i, j) in bonds_to_break:
+        glue_atoms.append(i)
+        glue_atoms.append(j)
+    graph = model["rigid_body_graph"]
+    members = model["rigid_body_members"]
+    if visited is None:
+        visited = []
+    nr_atoms = len(members[seed_node])
+    compensation = -1  # first atom after rotatable bond doesn't move, so doesn't weigh
+    # atoms in breakable bonds count twice, because of added complexity
+    compensation += int(sum([i in glue_atoms for i in members[seed_node]]))
+    this_value = depth**2 * (nr_atoms + compensation)
+    max_value = this_value
+    visited.append(seed_node)
+    for node in graph[seed_node]:
+        if node not in visited:
+            visited.append(node)
+            new_value = _calc_max_weighted_depth(model, node, bonds_to_break, visited, depth + 1)
+            max_value = max(max_value, new_value + this_value)
+    return max_value
+
+
+def merge_terminal_atoms(flex_model, not_terminal_atoms=()):
+    members_dict = flex_model["rigid_body_members"]
+    graph = flex_model["rigid_body_graph"]
+    remove = {}
+    for group_index, members in members_dict.items():
+        if (
+            len(members) == 1
+            and len(graph[group_index]) == 1
+            and members[0] not in not_terminal_atoms
+            and group_index != flex_model["root"]
+        ):
+            remove[group_index] = members[0]
+    flex_model["rigid_body_count"] -= len(remove)
+    for group_index, atom_index in remove.items():
+        host_index = graph[group_index][0]
+        flex_model["rigid_body_members"].pop(group_index)
+        flex_model["rigid_body_members"][host_index].append(atom_index)
+        flex_model["rigid_body_graph"].pop(group_index)
+        flex_model["rigid_body_graph"][host_index].remove(group_index)
+        flex_model["rigid_body_connectivity"].pop((host_index, group_index))
+        flex_model["rigid_body_connectivity"].pop((group_index, host_index))
+        flex_model["rigid_index_by_atom"][atom_index] = host_index
+    return
+
+
+def get_flexibility_model(
+    molsetup,
+    root_atom_index=None,
+    break_combo_data=None,
+    glue_pseudo_atoms=None,
```

Comment:
glue_pseudo_atoms is not used anywhere in this function, should it be deleted or should we keep it for future functionality

Reply:
Will remove, it's only needed in `update_closure_atoms` where it's called `coords_dict` (will rename to `glue_pseudo_atoms` for consistency). Just for reference here's a test for this feature: https://github.com/forlilab/Meeko/blob/6317966790b4e9817e37649bf771adb13bc3d921/test/macrocycle_test.py#L35

Reply:
removed in ac2da0724891385abdebe0a207e7a1fb7a202453